src/main/scala/com/datastax/gatling/plugin/response/DseResponseHandler.scala

```diff
@@ -35,20 +35,26 @@ object DseResponseHandler {
     .mkString(",")
 }
 
-abstract class DseResponseHandler[RS, Response <: DseResponse] extends StrictLogging with FutureCallback[RS] {
+trait DseResponseCallback[RS] {
+  def onFailure(t: Throwable): Unit
+
+  def onSuccess(result: RS): Unit
+}
```

Comment:
We were using Guava's FutureCallback here but not as an attachment to a ListenableFuture; we were calling the callback methods directly.  In an effort to avoid introducing Guava if it wasn't necessary I shifted this around to a simple trait defining something similar to FutureCallback.

This change plus the removal of the use of Guava Function in the 1.x API (around transforming graph results) allowed me to avoid introducing a Guava dependency entirely.

Reply:
Can we just use standard scala `Future`?

Reply:
That's the thing; this isn't really used as a future in any sense.  It's just an encapsulation of a couple methods to call from within the request handling actor:

```scala
case class RecordResult[T](t: Try[T], callback: DseResponseCallback[T])

object DseRequestActor extends StrictLogging {
  def recordResult[T](result: RecordResult[T]): Unit = result match {
    case RecordResult(t, callback) => t match {
      case Success(resultSet) => callback.onSuccess(resultSet)
      case Failure(exception: ExecutionException) => callback.onFailure(exception.getCause)
      case Failure(exception: Exception) => callback.onFailure(exception)
      case Failure(exception: Throwable) =>
        logger.error("Caught an unexpected error, please file a ticket", exception)
        callback.onFailure(exception)
    }
  }
}
```

There is a case for re-using one of the future callback APIs here, but it just seemed excessively confusing to me.  We're just defining a simple callback here... seemed worth calling that out in the code.